EIPS/eip-3712.md

```diff
@@ -0,0 +1,336 @@
+
+## Specification
+
+Interfaces: eip.sol
+```solidity
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+interface IEIP {
+    
+    function name(uint256 id) external view returns (string memory);
+    
+    function symbol(uint256 id) external view returns (string memory);
+    
+    function decimals(uint256 id) external view returns (uint8);
+    
+    function totalSupply(uint256 id) external view returns (uint256);
+
+    function balanceOf(uint256 id,address account) external view returns (uint256);
+
+    function transfer(uint256 id,address recipient, uint256 amount) external returns (bool);
+    
+    function transferBatch(uint256[] memory ids,address recipient,uint256[] memory amounts)external returns(bool);
+    
+    function transferBatchMul(uint256[] memory ids,address[]memory recipient,uint256[] memory amounts)external returns(bool);
+
+    function allowance(uint256 id,address owner, address spender) external view returns (uint256);
+
+    function approve(uint256 id,address spender, uint256 amount) external returns (bool);
+    
+    function approve(address spender,bool status) external returns(bool);
+    
+    function transferFromBatch(uint256[] memory ids,address sender,address recipient,uint256[] memory amounts)external returns(bool);
+    
+    function transferFromBatchMul(uint256[] memory ids,address[]memory sender,address[]memory recipient,uint256[] memory amounts)external returns(bool);
+
+    function transferFrom(uint256 id,address sender, address recipient, uint256 amount) external returns (bool);
+
+    event Transfer(uint256 id,address indexed from, address indexed to, uint256 value);
+
+    event Approval(uint256 id,address indexed owner, address indexed spender, uint256 value);
+    
+    event TransferBatch(uint256[] ids,address indexed from,address indexed to,uint256[] value);
+    
+    event TransferBatchNoIndexed(uint256[] ids,address[] from,address[] to,uint256[] value);
+    
+    event ApproveBatch(uint256[] ids,address indexed from,address indexed to,uint256[] value);
+    
+    event ApproveBatchNoIndexed(uint256[] ids,address[] from,address to,uint256[] value);
+}
+```
+
+Each of the interfaces `totalSupply`,`balanceOf`,`transfer`,`allowance`,`transferFrom`,`approve(uint256 id,address spender, uint256 amount)`  introduces a new parameter id  in the [ERC-20](./eip-20.md)'s corresponding interface's parameter list to specify a fungible token's type. 
+
+
+`approve(address spender,bool status)` is similar to [ERC-1155](./eip-1155.md)'s interface. I t allows an approval to be set uniformly for all types of tokens.
+
+`transferBatch` and `transferFromBatch` remove the parameter bytes memory data that the [ERC-1155](./eip-1155.md) standard uses and implement batch transactions.
```

Comment:
Not a strict requirement, but I think it would be better to define how these functions work without reference to EIP-1155.

Reply:
thanks for your advice.